source/extensions/common/wasm/context.h

```diff
@@ -57,7 +65,8 @@ class Context : public Logger::Loggable<Logger::Id::wasm>,
                 public Http::StreamFilter,
                 public Network::ConnectionCallbacks,
                 public Network::Filter,
-                public std::enable_shared_from_this<Context> {
+                public std::enable_shared_from_this<Context>,
+                public Filters::Common::Expr::BaseWrapper {
```

Comment:
Use composition rather than subclassing.  The Context should contain what it needs from CELL rather than being a CELL object.

Reply:
Yeah, I can use the same pattern as WasmStateWrapper.